homeassistant/components/squeezebox/media_player.py

```diff
@@ -34,15 +35,19 @@
     STATE_OFF,
     STATE_PAUSED,
     STATE_PLAYING,
-    STATE_UNAVAILABLE,
 )
+from homeassistant.core import callback
 from homeassistant.helpers import config_validation as cv, entity_platform
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
 from homeassistant.util.dt import utcnow
 
-from .__init__ import start_server_discovery
 from .const import (
     DEFAULT_PORT,
+    DISCOVERY_TASK,
     DOMAIN,
     ENTRY_PLAYERS,
```

Comment:
We don't seem to use the entry players container.

Reply:
It's used by `async_unload_entry` in `__init__.py` to make sure that when we unload a server from the list of config entries, the associated players are removed from the list of known players too. Otherwise if we remove a server and then add it back (for example if IP address changes), it won't re-detect the known players without a restart.

Reply:
I think each entity should remove itself from the list inside `async_will_remove_from_hass` entity coroutine method instead.

Reply:
We're not allowed to access entities from outside its platform. Entities are a platform concern.

Reply:
This is much cleaner. Thanks.